jlm/hls/backend/rvsdg2rhls/ThetaConversion.cpp

```diff
@@ -21,13 +21,22 @@ ConvertThetaNode(rvsdg::ThetaNode & theta)
   // add loopvars and populate the smap
   for (size_t i = 0; i < theta.ninputs(); i++)
   {
-    jlm::rvsdg::output * buffer;
-    loop->add_loopvar(theta.input(i)->origin(), &buffer);
-    smap.insert(theta.input(i)->argument(), buffer);
-    // buffer out is only used by branch
-    branches.push_back(*buffer->begin());
-    // divert theta outputs
-    theta.output(i)->divert_users(loop->output(i));
+    if (theta.input(i)->result()->origin() == theta.input(i)->argument())
```

Comment:
https://github.com/phate/jlm/blob/bcd76d06c3bd9619343cb090cca5207a72168653/jlm/rvsdg/theta.hpp#L290

Reply:
Changed to is_invariant()